### PR TITLE
DAT-68012 [Timbermill2] Fail to process keys with dot char in key

### PR DESCRIPTION
**Pull request by oz-zafar**

### CODE REVIEW

File: timbermill-java/timbermill-client/src/main/java/com/datorama/oss/timbermill/unit/Event.java [hunk 1]
```diff
@@ -322,7 +322,7 @@ public void trimAllStrings() {
 			Map<String, Number> newMetrics = Maps.newHashMap();
 			for (Map.Entry<String, Number> entry : metrics.entrySet()) {
 				Number value = entry.getValue();
-				String key = entry.getKey();
+				String key = entry.getKey().replace(".", "_");
 				if (value != null) {
```
Comment (oz-zafar) [inline timbermill-java/timbermill-client/src/main/java/com/datorama/oss/timbermill/unit/Event.java:326]: actually, I saw we already do that in getTrimmedLongValues to all fields **beside** metric:
		strings = getTrimmedLongValues(strings, Constants.STRING);
		context = getTrimmedLongValues(context, Constants.CTX);
		text = getTrimmedLongValues(text, Constants.TEXT);
		metrics = removeNaNs();
maybe was there a reson?

File: timbermill-java/timbermill-local/src/main/java/com/datorama/oss/timbermill/common/disk/IndexRetryManager.java [hunk 2]
```diff
@@ -79,14 +79,13 @@ public int retrySendDbBulkRequest(DbBulkRequest dbBulkRequest, BulkResponse resp
 			}
 		}
 		// finishing to retry - if persistence is defined then try to persist the failed requests
-		LOG.error(FLOW_ID_LOG + " Bulk #{} Reached maximum tries ({}) attempt to index.", flowId, bulkNum, numOfElasticSearchActionsTries);
+		LOG.warn(FLOW_ID_LOG + " Bulk #{} Reached maximum tries ({}) attempt to index.{}", flowId, bulkNum, numOfElasticSearchActionsTries, hasPersistence()? " Bulk will be persist to disk":"");
 		tryPersistBulkRequest(dbBulkRequest, flowId, bulkNum);
```
Comment (oz-zafar) [inline timbermill-java/timbermill-local/src/main/java/com/datorama/oss/timbermill/common/disk/IndexRetryManager.java:83]: not related to the task

Comment (oz-zafar) [inline timbermill-java/timbermill-local/src/main/java/com/datorama/oss/timbermill/common/disk/IndexRetryManager.java:83]: it's not related to the task, 
I changed a little bit the logs of the db

File: timbermill-java/timbermill-server/src/main/java/com/datorama/timbermill/server/service/TimbermillService.java [hunk 3]
```diff
@@ -63,7 +63,7 @@ public TimbermillService(@Value("${INDEX_BULK_SIZE:200000}") Integer indexBulkSi
 			@Value("${DELETION_CRON_EXPRESSION:0 0 12 1/1 * ? *}") String deletionCronExp,
 			@Value("${DELETION_CRON_MAX_INDICES_IN_PARALLEL:1}") int expiredMaxIndicesToDeleteInParallel,
 			@Value("${DISK_HANDLER_STRATEGY:sqlite}") String diskHandlerStrategy,
-			@Value("${BULK_PERSISTENT_FETCH_CRON_EXPRESSION:0 0/10 * 1/1 * ? *}") String bulkPersistentFetchCronExp,
+			@Value("${BULK_PERSISTENT_FETCH_CRON_EXPRESSION:30 2/10 * 1/1 * ? *}") String bulkPersistentFetchCronExp,
 			@Value("${EVENTS_PERSISTENT_FETCH_CRON_EXPRESSION:0 0/5 * 1/1 * ? *}") String eventsPersistentFetchCronExp,
```
Comment (oz-zafar) [inline timbermill-java/timbermill-server/src/main/java/com/datorama/timbermill/server/service/TimbermillService.java:67]: Dvir asked me to change the "failed bulks" cron to not time-overlap the "overflowed events" cron, in order to check if is it the cause to the latest errors in sqlite:
image

12:02:30,12:12:30,12:22:30,12:32:30,12:42:30
12:00:00,12:05:00,12:10:00,12:15:00,12:20:00

Comment (oz-zafar) [reply timbermill-java/timbermill-server/src/main/java/com/datorama/timbermill/server/service/TimbermillService.java]: nvm - saw you already changed it

File: timbermill-java/timbermill-client/src/main/java/com/datorama/oss/timbermill/unit/Event.java (hunk 1 above)
Comment (edengal) [reply timbermill-java/timbermill-client/src/main/java/com/datorama/oss/timbermill/unit/Event.java:326]: No reason as far as I can see.

File: timbermill-java/timbermill-local/src/main/java/com/datorama/oss/timbermill/common/disk/IndexRetryManager.java (hunk 2 above)
Comment (edengal) [reply timbermill-java/timbermill-local/src/main/java/com/datorama/oss/timbermill/common/disk/IndexRetryManager.java:83]: Let's keep it an ERROR, I want to get an alert when we persist to disk.

File: timbermill-java/timbermill-client/src/main/java/com/datorama/oss/timbermill/unit/Event.java (hunk 1 above)
Comment (edengal) [reply timbermill-java/timbermill-client/src/main/java/com/datorama/oss/timbermill/unit/Event.java:326]: Let's put it all in one place.

Comment (oz-zafar) [reply timbermill-java/timbermill-client/src/main/java/com/datorama/oss/timbermill/unit/Event.java:326]: hope I understood it right

**Review by edengal — APPROVED** [review]

**Pull request merged**